src/main/java/com/autotune/analyzer/recommendations/engine/DurationBasedRecommendationEngine.java

```diff
@@ -65,17 +69,18 @@ public AnalyzerConstants.RecommendationCategory getEngineCategory() {
 
     @Override
     public HashMap<String, Recommendation> getRecommendations(ContainerData containerData, Timestamp monitoringEndTime) {
-        // TODO: Needs to be implemented
-        AnalyzerConstants.RecommendationCategory recommendationCategory = AnalyzerConstants.RecommendationCategory.DURATION_BASED;
+        // Get the results
         HashMap<Timestamp, IntervalResults> resultsMap = containerData.getResults();
-        Timestamp minDate = resultsMap.keySet().stream().min(Timestamp::compareTo).get();
+        // Create a
```

Comment:
Incomplete comment

Reply:
Thanks @dinogun for the review, Made changes accordingly